src/components/calculator/SavedDataSelector.tsx

```diff
@@ -48,19 +65,47 @@ export const SavedDataSelector: React.FunctionComponent<{
         <Typeahead
           clearButton={true}
           emptyLabel={t('calculator.no_prebuilt_scenario_found')}
-          multiple
+          filterBy={showMatchesAndCustomScenario}
           highlightOnlyResult={true}
           id="predefined-typeahead"
-          inputProps={{ autoComplete: 'chrome-off' }}
+          inputProps={{
+            autoComplete: 'chrome-off',
+            shouldSelectHint: (shouldSelect: boolean, event) => {
+              if (event.keyCode === 13 /* return */) {
+                return true
+              }
+              return shouldSelect
+            },
+          }}
           onChange={(e: string[]) => {
             if (e.length === 0) {
               setSavedData('')
             }
             setSavedData(e[e.length - 1])
           }}
+          onBlur={() => {
+            if (selected2.length === 0) {
+              selected2 = [i18n.t('scenario.custom')]
+              if (savedDataRef.current) {
+                savedDataRef.current.getInput().value = i18n.t(
```

Comment:
I think this is unnecessary because setSavedData triggers a re-render, which will set the value via the `selected` prop

Reply:
Done and tested